clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -250,22 +250,24 @@ def ConstPtrAttr : CIR_Attr<"ConstPtr", "ptr", [TypedAttrInterface]> {
   let summary = "Holds a constant pointer value";
   let parameters = (ins
     AttributeSelfTypeParameter<"", "::mlir::cir::PointerType">:$type,
-    "uint64_t":$value);
+    "mlir::IntegerAttr":$value);
   let description = [{
     A pointer attribute is a literal attribute that represents an integral
     value of a pointer type.
   }];
   let builders = [
-    AttrBuilderWithInferredContext<(ins "Type":$type, "uint64_t":$value), [{
-      return $_get(type.getContext(), type.cast<mlir::cir::PointerType>(), value);
+    AttrBuilderWithInferredContext<(ins "Type":$type, "int64_t":$value), [{
```

Comment:
I suggest we remove this functionality, better if we force whomever is creating the attribute to pass in the `mlir::IntegerType`, assumptions might lead to subtle mistakes here.

Reply:
updated